atd-toolbox/data_model/python/cris_import.py

```diff
@@ -0,0 +1,427 @@
+    return rows
+
+
+def set_default_values(records, key_values):
+    for record in records:
+        for key, value in key_values.items():
+            record[key] = value
+
+
+def combine_date_time_fields(
+    rows, *, date_field_name, time_field_name, is_am_pm_format
+):
+    """Combine a date and time field and format as ISO string. The same ISO
+    string is stored in both input field names.
+    """
+    tzinfo = ZoneInfo("America/Chicago")
+    for row in rows:
+        input_date_string = row[date_field_name]
+        input_time_string = row[time_field_name]
+
+        if not input_date_string or not input_time_string:
+            continue
+
+        # parse a date string that looks like '12/22/2023'
+        month, day, year = input_date_string.split("/")
+
+        hour = None
+        minute = None
+        if is_am_pm_format:
+            # parse a time string that looks like '12:15 PM'
+            hour_minute, am_pm = input_time_string.split(" ")
+            hour, minute = hour_minute.split(":")
+            if am_pm.lower() == "pm":
+                hour = int(hour) + 12 if hour != "12" else int(hour)
+            else:
+                hour = int(hour) if hour != "12" else 0
+        else:
+            # parse a time string that looks like '12:00:00'
+            hour, minute, second = input_time_string.split(":")
+
+        # create a tz-aware instance of this datetime
+
+        dt = datetime(
+            int(year),
+            int(month),
+            int(day),
+            hour=int(hour),
+            minute=int(minute),
+            tzinfo=tzinfo,
+        )
+        # save the ISO string with tz offset
+        crash_date_iso = dt.isoformat()
+        row[date_field_name] = crash_date_iso
+        row[time_field_name] = crash_date_iso
+
+
+def chunks(lst, n):
+    """Yield successive n-sized chunks from lst."""
+    for i in range(0, len(lst), n):
+        yield lst[i : i + n]
+
+
+def set_peh_field(record):
+    """Determine if a person was experiencing homeless based on
+    reportin conventions used by law enforcement crash investigator"""
+    for field in peh_fields:
+        record_val = record[field]
+        for search_term in ["homeless", "unhoused", "transient"]:
+            if record_val and search_term in record_val.lower():
+                record["prsn_exp_homelessness"] = True
+                # no need to continue further
+                return
+
+
+def nullify_name_fields(records):
+    for record in records:
+        if record["prsn_injry_sev_id"] != "4":
+            for field in name_fields:
+                record[field] = None
+    return
+
+
+def main():
+    overall_start_tme = time.time()
+
+    print("downloading column metadata...")
+    column_metadata = load_data(COLUMN_ENDPOINT)
+
+    files_todo = get_files_todo()
+
+    extract_ids = list(set([f["extract_id"] for f in files_todo]))
+    # assumes extract ids are sortable oldest > newest by filename. todo: is that right?
```

Comment:
I'm not sure this is true. Here are some extracts which I recently pulled:

<img width="1477" alt="image" src="https://github.com/cityofaustin/atd-vz-data/assets/3653206/331de972-1108-4e52-8a8c-17f7df29af80">

Because CRIS puts a schema number in the second element of the name, I think it becomes undetermined, as an extract commonly can have multiple sets of CSVs in them for when the data spans CRIS schemas. The following element is the date of the request, so you 'mostly' get them in the order you requested when you sort, but not reliably. 

While I think the assertion that you can't count on the order to be useful is true, the nitty-gritty details are always hard to pin down on CRIS, so this comment is YMMV for better or worse.

Reply:
>  because the third element of the CSVs' filename is the date stamp of the extract request, so sorting them will just get back the order they were requested / received from CRIS

thanks—this is the assumption i had made so it's good to hear you confirm. it's not clear from this comment, but my concern was indeed to make sure that we always process extracts in the order they were processed by CRIS. if we process extracts in the wrong order, we could theoretically update a crash record with old data.